sql-plugin/src/main/scala/com/nvidia/spark/rapids/TypeChecks.scala

```diff
@@ -766,7 +766,7 @@ class CastChecks extends ExprChecks {
   val mapChecks: TypeSig = none
   val sparkMapSig: TypeSig = STRING + MAP.nested(all)
 
-  val structChecks: TypeSig = none
+  val structChecks: TypeSig = STRING
```

Comment:
We probably need some updates to how this checks nested types, because this currently says that we can cast a struct of anything to a string, and that is not technically true.  We are going to need to be recursive in how we check because we still want things like floats being cast to strings only working if we have enabled it.

https://github.com/NVIDIA/spark-rapids/blob/de00b8a238201334bd59bedcebb76e8d5022e2c2/sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsConf.scala#L530

Reply:
I've added recursive checks to the struct type checking when tagging the expression for GPU. `TypeChecks.scala` definition and `supported_ops.md` documentation will continue to believe that struct to string casting has blanket support -- should there be an extra note in the `supported_ops.md` documentation?